src/train_lm.py

```diff
@@ -308,12 +308,16 @@ def compute_objective(metrics: Dict[str, float]) -> float:
     # Initialize our Trainer
     trainer_cls = TrainerDeterministicSampler if training_args.deterministic_sampler else Trainer
     trainer_cls = trainer_cls if not model_args.seq2seq else Seq2SeqTrainer
+
+    if training_args.do_eval is False:
```

Comment:
@egorkrash I remember you had some runs failing because of smth related to this in your branch, do you remember what it was?

Reply:
they just didn't do evals at all, not failing

Reply:
because you replace eval set with {} and so callbacks also do not work when do eval = False

Reply:
Ah I see. So currently we cannot disable the built in callback for loss evaluation while not also disabling your grad alignment callback?

Reply:
I don't understand why the previous version didn't work. We had eval_dataset=None when do_eval=False and the model should never execute eval loop. But somehow it reached eval loop and failed

Reply:
Yeah I remember, plausibly this was a HF bug

Reply:
> Ah I see. So currently we cannot disable the built in callback for loss evaluation while not also disabling your grad alignment callback?

@egorkrash?

Reply:
Currently we cannot